doc/third-party-go-clients.md

```diff
@@ -41,64 +47,95 @@ connr, err := connector.NewSimple(connector.SimpleConfig{
-specified).
-
 ### `ClientStoreConnector`
 
 If a client store is available on the filesystem, it is possible to use it to
 configure the connector.
 
+The NewClientStore will try to find a client store at the location specified by
+the JUJU_DATA environment variable which defaults to `~/.local/share/juju`. It is
+also possible to pass in a custom value(see `clientstoreconnector.go` for details).
+
 ```golang
 import "github.com/juju/juju/api/connector"
 ```
 ```golang
-connr, err := connector.NewClientStore(connector.ClientStoreConfig{
-    ControllerName: "overlord",
-})
+    connr, err := connector.NewClientStore(connector.ClientStoreConfig{
+        ControllerName: "overlord",
+    })
+    if err != nil {
+        log.Fatalf("Error getting new juju connector: %s", err)
+    }
 ```
 
-The above will try to find a client store in the default location.  It is also
-possible to pass in a custom value (see `clientstoreconnector.go` for details).
+## Make a connection
 
-## Making requests
+Once we have a connector, make a connection. The connection can be used by multiple
+api clients before being closed. It is essential to close the connection when finished
+using.
 
-Once we have a connector, it's easy to make requests.
+```golang
+    // Get a connection
+    conn, err := connr.Connect()
+    if err != nil {
+        log.Fatalf("Error opening connection: %s", err)
+    }
+    defer func(){ _ = conn.Close() }()
+````
+
+## Make requests via an api client
+
+Connections can be used to get a juju api facade client. The client is then used to make
+api calls. One connection can be used by multiple api clients.
 
 ```golang
 import (
-    "encoding/json"
-	"github.com/juju/juju/api/connector"
-	"github.com/juju/juju/api/client/client"
+	"fmt"
+	
+	"github.com/juju/juju/api/client/modelmanager"
 )
 ```
 ```golang
-    // Get a connection
-	conn, err := connr.Connect()
-	if err != nil {
-		log.Fatalf("Error opening connection: %s", err)
-	}
-    defer conn.Close()
+    // Get a model manager facade client.
+	client := modelmanager.NewClient(conn)
 
-    // Get a Client facade client
-	client := apiclient.NewClient(conn)
-
-    // Call the Status endpoint of the client facade
-    status, err := client.Status(nil)
+    // Find model info for the controller's models.
+    info, err := client.ListModels("admin")
     if err != nil {
-        log.Fatalf("Error requesting status: %s", err)
+        log.Fatalf("Error requesting model info: %s", err)
     }
 
+	// Print a list of all model names.
+	names := make([]string, len(info))
+	for i, model := range info {
+		names[i] = model.Name
+    }
     // Print to stdout.
-    b, err := json.MarshalIndent(status, "", "  ")
-	if err != nil {
-		log.Fatalf("Error marshalling response: %s", err)
-	}
-	fmt.Printf("%s\n", b)
+	fmt.Printf("%s\n", strings.Join(names, ", "))
 ```
 
-## Points to address
+## Available Juju client facades
+
+The juju client apis are located in `api/client` directory.
+
```

Comment:
This feels like good content for a doc.go file under https://github.com/juju/juju/tree/3.6/api/client . We can then just link to that from the relevant step above.

PS Actually, having taken a quick look at that directory, I see a lot of other items there in addition to the ones you list here. Wondering why the lists are different.

Reply:
This doc references packages in many different files. Not sure it belongs in api/client.